pkg/ctl/cmdutils/filter.go

```diff
@@ -0,0 +1,205 @@
+			return mustInclude
+		}
+	}
+
+	if hasExcludeRules {
+		exclude := f.excludeNames.Has(name)
+		if f.matchGlobs(name, f.excludeGlobs) {
+			exclude = true
+		}
+		if exclude && !mustInclude {
+			// exclude, unless overridden by an inclusion rule
+			return false
+		}
+	}
+
+	return true // biased to include
+}
+
+// doMatchAll all names against the filter and return two sets of names - included and excluded
+func (f *Filter) doMatchAll(names []string) (sets.String, sets.String) {
+	included, excluded := sets.NewString(), sets.NewString()
+	if f.ExcludeAll {
+		for _, n := range names {
+			excluded.Insert(n)
+		}
+		return included, excluded
+	}
+	for _, n := range names {
+		if f.Match(n) {
+			included.Insert(n)
+		} else {
+			excluded.Insert(n)
+		}
+	}
+	return included, excluded
+}
+
+// doAppendIncludeGlobs sets globs for inclusion rules
+func (f *Filter) doAppendIncludeGlobs(names []string, resource string, globExprs ...string) error {
+	for _, expr := range globExprs {
+		compiledExpr, err := glob.Compile(expr)
+		if err != nil {
+			return errors.Wrapf(err, "parsing glob filter %q", expr)
+		}
+		f.includeGlobs = append(f.includeGlobs, compiledExpr)
+		f.rawIncludeGlobs = append(f.rawIncludeGlobs, expr)
+	}
+	return f.includeGlobsMatchAnything(names, resource)
+}
+
+func (f *Filter) doSetExcludeExistingFilter(names []string, resource string) error {
+	f.excludeNames.Insert(names...)
+	for _, n := range names {
+		isAlsoIncluded := f.includeNames.Has(n)
+		if f.matchGlobs(n, f.includeGlobs) {
+			isAlsoIncluded = true
+		}
+		if isAlsoIncluded {
+			return fmt.Errorf("existing %s %q should be excluded, but matches include filter: %s", resource, n, f.describeIncludeRules())
+		}
+	}
+	return nil
+}
+
+
+func (f *Filter) includeGlobsMatchAnything(names []string, resource string) error {
+	if len(f.includeGlobs) == 0 {
+		return nil
+	}
+	for _, n := range names {
+		if f.matchGlobs(n, f.includeGlobs) {
+			return nil
+		}
+	}
+	return fmt.Errorf("no %ss match include glob filter specification: %q", resource, strings.Join(f.rawIncludeGlobs, ","))
+}
+
+func (f *Filter) doLogInfo(resource string, names []string) {
+	logMsg := func(subset sets.String, status string) {
+		count := subset.Len()
+		list := strings.Join(subset.List(), ", ")
+		subjectFmt := "%d %ss (%s) were %s"
+		if count == 1 {
+			subjectFmt = "%d %s (%s) was %s"
+		}
+		logger.Info(subjectFmt, count, resource, list, status)
+	}
+
+	included, excluded := f.doMatchAll(names)
+	if f.hasIncludeRules() {
+		logger.Info("include rules: %s", f.describeIncludeRules())
+		if included.Len() == 0 {
+			logger.Info("no %ss were included by the filter", resource)
+		}
+	}
+	if included.Len() > 0 {
+		logMsg(included, "included")
+	}
+	if f.hasExcludeRules() {
+		logger.Info("exclude rules: %s", f.describeExcludeRules())
+		if excluded.Len() == 0 {
+			logger.Info("no %ss were excluded by the filter", resource)
+		}
+	}
+	if excluded.Len() > 0 {
+		logMsg(excluded, "excluded")
+	}
+}
```

Comment:
Über nit: elsewhere in this commit you added a new line, so I'm guessing you may want to add one here too in order to be consistent?

Reply:
No :)